baseline-error-prone/src/main/java/com/palantir/baseline/errorprone/IllegalSafeLoggingArgument.java

```diff
@@ -260,11 +263,24 @@ public Description matchReturn(ReturnTree tree, VisitorState state) {
         while (path != null && path.getLeaf() instanceof StatementTree) {
             path = path.getParentPath();
         }
-        if (path == null || !(path.getLeaf() instanceof MethodTree)) {
+        if (path == null || !(path.getLeaf() instanceof MethodTree || path.getLeaf() instanceof LambdaExpressionTree)) {
             return Description.NO_MATCH;
         }
-        MethodTree method = (MethodTree) path.getLeaf();
+
+        if (path.getLeaf() instanceof MethodTree method) {
```

Comment:
style nit: we can combine these two sets of conditionals together to get rid of the `throw new IllegalStateException` since it should be impossible.

```java
if (path != null) {
    if (path.getLeaf() instanceof MethodTree method) {
        // etc
    } else if (path.getLeaf() instanceof LambdaExpressionTree lambda) {
        // etc
    }
}
return Description.NO_MATCH;

Reply:
Oh yeah, that's much nicer